control_panel_api/pagination.py

```diff
@@ -1,5 +1,37 @@
-from rest_framework.pagination import PageNumberPagination
+from rest_framework.pagination import PageNumberPagination, _positive_int
 
 
 class CustomPageNumberPagination(PageNumberPagination):
+    """This Pagination class allows a request to pass the page_size query param
+    for usual pagination functionality but adds custom whereby a value of
+    page_size=0 will return all items
+    """
     page_size_query_param = 'page_size'
+
+    def paginate_queryset(self, queryset, request, view=None):
+        """Override to check for page_size=0 if so then return all items else
+        execute original parent functionality
+        """
+        page_size = self.get_page_size(request)
+        if page_size != 0:
+            return super().paginate_queryset(queryset, request, view)
+
+        paginator = self.django_paginator_class(queryset, queryset.count())
+        self.page = paginator.page(1)
```

Comment:
I'm not familiar with Django pagination, etc...so I may miss something here, isn't this going to return always page 1?

Reply:
Yep. Code is to always return the entire result set, which implies just one page which is 1.

Reply:
Sorry, didn't see the `return super()...`.

Reply:
😱

Reply:
Don't understand why Python devs hate if/else...

Reply:
That's not a python thing!  Avoids unnecessary nesting.

Reply:
😱

Reply:
![doge](https://cdn.theatlantic.com/assets/media/img/upload/wire/2013/12/31/doge-1/square.jpg?1473195192)